pkg/cluster/gitrepo/gitrepo_cluster.go

```diff
@@ -1126,6 +1195,306 @@ func (g *clusterGitRepo) UpdateTags(ctx context.Context, application, cluster, t
 	return nil
 }
 
+func (g *clusterGitRepo) DefaultBranch() string {
+	return g.defaultBranch
+}
+
+func (g *clusterGitRepo) UpgradeCluster(ctx context.Context,
+	param *UpgradeValuesParam) (string, error) {
+	const op = "cluster git repo: upgrade cluster"
+	defer wlog.Start(ctx, op).StopPrint()
+	currentUser, err := common.UserFromContext(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	pid := fmt.Sprintf("%v/%v/%v", g.clustersGroup.FullPath,
+		param.Application, param.Cluster)
+
+	type upgradeValueBytes struct {
+		fileName string
+		in       []byte
+		out      []byte
```

Comment:
I don't know what is in & out, make it more readable please

Reply:
fixed